mcodingbot/plugins/highlights.py

```diff
@@ -17,6 +18,22 @@
 MAX_HIGHLIGHT_LENGTH = 32
 
 
+async def has_permission(guild_id: int, user_id: int, channel_id: int) -> bool:
+    member = plugin.app.cache.get_member(guild_id, user_id)
+    if not member:
+        return False
+
+    channel = plugin.app.cache.get_guild_channel(channel_id)
+    if not channel:
+        # must be a thread
+        thread = plugin.app.rest.fetch_channel(channel_id)
```

Comment:
I think this needs to gracefully handle `hikari.NotFoundError` for me to be confident in the code